doc/source/ray-observability/monitoring-debugging/troubleshoot-failures.rst

```diff
@@ -1,44 +1,252 @@
+This section explains how to debug out-of-memory errors in Ray.
+
+**Step 1: Find tasks and actors that use high memory**
+
+First, find the tasks and actors that use high memory usage. View the :ref:`per task and actor memory usage graph <dash-workflow-cpu-memory-analysis>` for more details.
+The memory usage from the per component graph uses RSS - SHR. See the below for reasoning.
+
+Alternatively, you can also use the CLI command `htop <https://htop.dev/>`_.
+
+.. image:: ../images/htop.png
+    :align: center
+
+See the ``allocate_memory`` row. See two columns, RSS and SHR. 
+
+SHR usage is typically the memory usage from the Ray object store. The Ray object store allocates 30% of host memory to the shared memory (``/dev/shm``, unless you specify ``--object-store-memory``).
+If Ray workers access the object inside the object store using ``ray.get``, SHR usage increases. Since the Ray object store supports the :ref:`zero-copy <serialization-guide>`
+deserialization, several workers can access the same object without copying them to in-process memory. For example, if
+8 workers access the same object inside the Ray object store, each process' ``SHR`` usage increases. However, they are not using 8 * SHR memory (there's only 1 copy in the shared memory). 
+Also note that Ray object store triggers :ref:`object spilling <object-spilling>` when the object usage goes beyond the limit, which means the memory usage from the shared memory won't exceed 30%
+of the host memory.
+
+It means when there are out-of-memory issues from a host, it is due to RSS usage from each worker. We recommend to calculate per
+process memory usage by RSS - SHR because SHR is for Ray object store as explained above. That said, the total memory usage is typically
+``SHR (object store memory usage, 30% of memory) + sum(RSS - SHR from each ray proc) + sum(RSS - SHR from system components. e.g., raylet, GCS. Usually small)``.
+
+**Step 2 Debug memory problems, head node**
+
+First, check the head node memory usage from the metrics page. Find the head node address from the cluster page.
+
+.. image:: ../images/head-node-addr.png
+    :align: center
+
+And then check the memory usage from the head node from the node memory usage view inside the Dashboard :ref:`metrics view <dash-metrics-view>`.
+
+.. image:: ../images/metrics-node-view.png
+    :align: center
+
+Ray head node has more memory-demanding system components such as GCS or the dashboard. 
+Also, driver runs from a head node by default. If the head node has the same memory capacity as worker nodes
+and if you execute the same number of tasks and actors from a head node, it can easily have out-of-memory problems.
+In this case, we recommend you to not run any tasks and actors to the head node. This is achievable by specifying ``--num-cpus=0`` when starting a head node by ``ray start --head``.
```

Comment:
kuberay user won't run `ray start` directly. You can link to https://docs.ray.io/en/latest/ray-core/scheduling/resources.html#specifying-node-resources which shows different ways of specifying node resources.